static/bidder-params/lemmadigital.json

```diff
@@ -12,8 +12,13 @@
         "aid": {
             "type": "integer",
             "description": "Ad ID"
+        },
+        "host": {
+            "type": "string",
+            "description": "Network Host to request from",
+            "enum": ["uses", "usws", "usct", "emea", "sg", "doohsg", "doohus"]
```

Comment:
Good catch @SyntaxNode . @lm-ved , please see https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html#consider-your-geography and the example . Basically, pubs can't reasonably determine the geography.

```
# We have the following regional endpoint domains: "uses" (us-east), "usws" (us-west), "usct" (us-central), "emea" (europe middle east africa), "sg" (where's this?)
# Please deploy this config in each of your datacenters with the appropriate regional subdomain
endpoint: "http://REGION.example.com/openrtb2"
geoscope:
  - USA
  - CAN
```
As for separate endpoints for DOOH vs display, I think that's a problem. @SyntaxNode - can we allow them to create a bidder-specific "regioncode" YAML entry that's filled in by the host company and is available to their runtime code? If so, then the adapter can look at regioncode along with site/app/dooh and determine the endpoint.

Reply:
@SyntaxNode / @bretg ,
Would I be correct in assuming the following change would suffice in this adapter's case?
```
# We have the following regional endpoint domains: "uses" (us-east), "usws" (us-west), "usct" (us-central), "emea" (europe middle east africa), "sg" (singapore)
# Please deploy this config in each of your datacenters with the appropriate regional subdomain
endpoint: "https://REGION.ads.lemmatechnologies.com/lemma/servad?src=prebid&pid={{.PublisherID}}&aid={{.AdUnit}}"
geoscope:
  - global
```
instead of the current:
```
endpoint: "https://{{.Host}}.ads.lemmatechnologies.com/lemma/servad?src=prebid&pid={{.PublisherID}}&aid={{.AdUnit}}"
```
It also would depend on the decision regarding how to handle the different endpoints for DOOH.

Reply:
> @SyntaxNode - can we allow them to create a bidder-specific "regioncode" YAML entry that's filled in by the host company and is available to their runtime code?

Yup. This is a good use for the `extra_info` field. Hosts can override this value in the appropriate data centers just as they would the endpoint.

> Would I be correct in assuming the following change would suffice in this adapter's case?

Yes, but please disable this adapter by default (hosts can enable it) since the endpoint is not valid by default. This is the same approach used by the Rubicon/Magnite adapter.

Reply:
@SyntaxNode , thank you for the details.
I've added the host option as an `extra_info` field. I have also kept a default value and removed host from the `required` array of bidder-params. This would allow for both - a server-level config and a publisher to provide it if needed, with a default(set to the current adapter's host ) in case it is neither.

Reply:
> This would allow for both - a server-level config and a publisher to provide it if needed, with a default(set to the current adapter's host ) in case it is neither.

@bretg Are you good with giving publishers an **option** to set it per-request?

Reply:
I suppose. Don't expect anyone to get it right.